web-components/src/components/inputs/LocationStateField.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { Field } from 'formik';
+import { makeStyles } from '@mui/styles';
+import { Theme } from '../../theme/muiTheme';
+
+import SelectTextField, { Option } from './SelectTextField';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  textField: {
+    '& .MuiInputBase-formControl': {
+      marginTop: theme.spacing(2.25),
+    },
+  },
+}));
+
+interface FieldProps {
+  country: string;
+  triggerOnChange: (stateCountry: string) => void;
+}
+
+const LocationStateField: React.FC<FieldProps> = ({
+  country,
```

Comment:
we probably want to add the form field `name` as optional here, in case we want to use this field with a different field name than `stateProvince`